### PR TITLE
IGNITE-14866: add extra tests for functions

### PR DESCRIPTION
**Pull request by ygerzhedovich**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/test/sql/function/string/test_space.test [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# name: test/sql/function/string/space.test
+# description: Test the SPACE function
+# group: [string]
+
+query II
+select sha1('hello'), sha1(NULL)
```
Comment (korlov42) [inline modules/calcite/src/test/sql/function/string/test_space.test:6]: this query belongs to `sha1.test` script

File: modules/calcite/src/test/sql/function/string/test_strcmp.test [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# name: test/sql/function/string/test_strcmp.test
+# description: Test the md5 function
```
Comment (korlov42) [inline modules/calcite/src/test/sql/function/string/test_strcmp.test:2]: wring description

File: modules/calcite/src/test/sql/function/timestamp/test_timestampadd.test [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+query T
+SELECT TIMESTAMPADD(QUARTER, 1, TIMESTAMP '2021-12-31 08:22:33');
+----
+2022-03-31 08:22:33.0
+
+query T
+SELECT TIMESTAMPADD(YEAR, 1, TIMESTAMP '2021-12-31 08:22:33');
+----
+2022-12-31 08:22:33.0
+
```
Comment (korlov42) [inline modules/calcite/src/test/sql/function/timestamp/test_timestampadd.test:91]: too many empty rows at the end

File: modules/calcite/src/test/sql/function/string/test_space.test (hunk 1 above)
Comment (ygerzhedovich) [reply modules/calcite/src/test/sql/function/string/test_space.test]: fixed

File: modules/calcite/src/test/sql/function/string/test_strcmp.test (hunk 2 above)
Comment (ygerzhedovich) [reply modules/calcite/src/test/sql/function/string/test_strcmp.test]: fixed

File: modules/calcite/src/test/sql/function/timestamp/test_timestampadd.test (hunk 3 above)
Comment (ygerzhedovich) [reply modules/calcite/src/test/sql/function/timestamp/test_timestampadd.test]: fixed

**Review by korlov42 — APPROVED** [review]

**Pull request merged**